src/Metadata/Extractor/PhpDocClassExtractor.php

```diff
@@ -0,0 +1,105 @@
+<?php
+
+/*
+ * This file is part of the API Platform project.
+ *
+ * (c) Kévin Dunglas <dunglas@gmail.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace ApiPlatform\Metadata\Extractor;
+
+use PHPStan\PhpDocParser\Ast\PhpDoc\PhpDocNode;
+use PHPStan\PhpDocParser\Ast\PhpDoc\PhpDocTextNode;
+use PHPStan\PhpDocParser\Lexer\Lexer;
+use PHPStan\PhpDocParser\Parser\ConstExprParser;
+use PHPStan\PhpDocParser\Parser\PhpDocParser;
+use PHPStan\PhpDocParser\Parser\TokenIterator;
+use PHPStan\PhpDocParser\Parser\TypeParser;
+
+/**
+ * Extracts class data using a PHPDoc parser.
+ */
+final class PhpDocClassExtractor implements ClassDescriptionExtractorInterface
+{
+    private PhpDocParser $phpDocParser;
+    private Lexer $lexer;
+
+    /** @var array<string, PhpDocNode> */
+    private array $docBlocks = [];
+
+    public function __construct()
+    {
+        $this->phpDocParser = new PhpDocParser(new TypeParser(new ConstExprParser()), new ConstExprParser());
+        $this->lexer = new Lexer();
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getShortDescription(string $class, array $context = []): ?string
+    {
+        if (!$docBlock = $this->getDocBlock($class)) {
+            return null;
+        }
+
+        foreach ($docBlock->children as $docChild) {
+            if ($docChild instanceof PhpDocTextNode && !empty($docChild->text)) {
+                return $docChild->text;
+            }
+        }
+
+        return null;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getLongDescription(string $class, array $context = []): ?string
```

Comment:
We never need this right?

Reply:
I think it can be useful in OpenAPI and GraphQL.

Reply:
let's add this when we need it then we don't want to maintain dead code

Reply:
It's mostly to have the same interface as the Symfony one, but alright I will throw instead.